diff --git a/tools/alticonstool/alticonstool.py b/tools/alticonstool/alticonstool.py
--- a/tools/alticonstool/alticonstool.py
+++ b/tools/alticonstool/alticonstool.py
@@ -47,8 +47,12 @@ def insert_alticons(plist_data, alticons, device_families):
       "CFBundleIconFiles": sorted(set(map(iconname_from_filename, os.listdir(alticon)))),
     }
   if "iphone" in device_families:
+    if "CFBundleIcons" not in plist_data:
+      plist_data["CFBundleIcons"] = {}
     plist_data["CFBundleIcons"]["CFBundleAlternateIcons"] = alticons_data
   if "ipad" in device_families:
+    if "CFBundleIcons~ipad" not in plist_data:
+      plist_data["CFBundleIcons~ipad"] = {}
     plist_data["CFBundleIcons~ipad"]["CFBundleAlternateIcons"] = alticons_data
 
 
